### ISSUE TITLE
How could be the UI?

### ISSUE THREAD

**Issue by macabeus** [macabeus/klo-gba.js#11]
We need define how will be the UI for this project, which components should be create and how the user will interact with they.

In the first moment, I think that we could create something like it:

![image](https://user-images.githubusercontent.com/9501115/54504788-d9df3100-4913-11e9-9347-216b98703ad9.png)

We already have the components **1** (`Head`), **2** (`Map`) and **3** (`MapFooter`). The components **4** (`OAMTable?`), **5** (`Hexdump?`) and **6** (`FloatingMapOptions?`) should be create.

The `OAMTable` will list the OAMs on the map, showing the name, position, and other informations for each OAM. When a line is hover, the map should highlight it. When a table line is selected, the `Hexdump` should show informations about this OAM.
An important detail is that the game "recycle" the OAM at each vision step, then we need a friendly way to abstract it but we need keep the limits about it. Maybe slip this table in three sections, one section for step?

The `Hexdump` will be used to user can view and edit the informations about the OAM and (maybe also) the map. Would be nice if we can describe each byte, something like it:
![image](https://user-images.githubusercontent.com/9501115/54504666-486fbf00-4913-11e9-97fc-ccf90588f30e.png)
Another good example [is this web tool](http://hex-works.com/eng).
But, at the first moment, [we can just use this React component](https://github.com/travisperson/react-hexviewer), because it is already working.
I think that this hexdump component could be an isolated react component.

The `FloatingMapOptions` will be a floating and draggable box on map, that has some options to change the map display. When the mouse is far from it, should has a little opacity.
At the first moment, I think that nice options is "Show/hide OAM" and `Show/hide grid".

**Some questions:**
- which others components we can create to improve the UI/UX?
- which columns the `OAMTable` should have?
- how should be the highlight on map when a line at `OAMTable` is hovered?
- which options the `FloatingMapOptions` should have?

**Comment by GabrielaBezerra** [comment]
I guess this is already good, similar to another editors, but it would be better if the user had the option to show/hide all the components, and could change the amount of space that each component takes on the screen. Also, try not to give too much space for something that the user will not use very often... Depending on the amount of columns, giving up on the Hex screen, and placing the table as a sidebar might be a good idea.

**Comment by Deivis** [comment]
I'm not an expert in geme maps, but the `FloatingMapOptions`  could be side by in the head as an dropdown to avoid tools over the map.
Any tool which could be over the map  must have a minimize option so as not to disturb the map view.

**Comment by macabeus** [comment]
Okay. I liked the @Deivis suggestion. Then, I changed the `FloatingMapOptions` to a button with popover (`DisplayMapOptions`?).

Also, I added a new element, a simple GitHub button (`GitHubButton`?) at the left corner of head. Then, the title now will be at the center of the head.

![image](https://user-images.githubusercontent.com/9501115/54870914-7a59a900-4d8b-11e9-92eb-0f630e070d16.png)

**Comment by macabeus** [comment]
I added one new elements at the wireframe: `zoom`.

In the drawing, I added the `zoom` tool at `DisplayMapOptions`, because the zoom changes the displaying, but maybe a better place could be the bar above the tilemap.
We need to fix [this bug](https://github.com/macabeus/klo-gba.js/issues/14) before start to develop it, because a larger zoom will broke the UI.

![image](https://user-images.githubusercontent.com/9501115/55282047-35012280-531b-11e9-842d-b2696430cef6.png)

**Comment by macabeus** [comment]
More updates. I added two components to user can edit the level and updated **5** (`Hexdump`? `HexViewer`?).

![image](https://user-images.githubusercontent.com/9501115/56102430-6bd25d80-5f03-11e9-87ce-ed7ab0384d94.png)

The component **9** (`TileSet`?) is used to show the tiles that the user add on the level. One of this tiles should be the `Empty`, to clear a position. When the user select a tile at `TileSet`, the component **10** (`TileSelected`?) should be updated to display the informations about this tile (such as name, id, if it is blocker or non-blocking).

The component **5** (`Hexdump`? `HexViewer`?) was updated adding a tab bar, to switch between to display hexdump about OAM or about tilemap.
